vmware_guest_disk: add sharing (multi-writer) capability

cspeidel opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Closes #212 

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
vmware_guest_disk

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
@Akasurde This draft PR has an initial commit that I've successfully tested on vSphere client 7.0.0. This change allows you to set the multi-writer capability of a disk. Currently I have this option as a boolean, but let me know if you think it should be a string with some other values such as 'none' and 'multi_writer'. 
I would like some feedback on the above before writing the tests and examples.

The following YAML will create a disk with sharing set to multi-writer.

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below
- name: Add disks to virtual machine using name
  vmware_guest_disk:
    hostname: "{{ vcenter_hostname }}"
    username: "{{ vcenter_username }}"
    password: "{{ vcenter_password }}"
    datacenter: "{{ datacenter_name }}"
    validate_certs: no
    name: VM_225
    disk:
      - size_mb: 10
        type: eagerzeroedthick
        datastore: datacluster0
        state: present
        scsi_controller: 1
        unit_number: 1
        scsi_type: 'paravirtual'
        sharing: True
```

goneri commented:
Thanks for the PR @cspeidel, could you add a test case in https://github.com/ansible-collections/vmware/blob/master/tests/integration/targets/vmware_guest_disk/tasks/main.yml to cover the new option?

goneri commented on plugins/modules/vmware_guest_disk.py:
```diff
@@ -718,6 +725,16 @@ def sanitize_disk_inputs(self):
                                           " 'disk_mode' value from ['persistent', 'independent_persistent', 'independent_nonpersistent']." % disk_index)
             current_disk['disk_mode'] = temp_disk_mode
 
+            # Sharing mode of Disk
```
This function is already way to long, could you move this block out in a function. something like: `get_sharing()`.

cspeidel replied:
I refactored disk sharing into the get_sharing method with 855a1b305bc6e39ed5c1942753a01eba05e89236. I'll proceed to work on the test cases.

goneri commented on plugins/modules/vmware_guest_disk.py:
```diff
@@ -416,6 +421,26 @@ def get_ioandshares_diskconfig(self, disk_spec, disk):
+            sharing_mode: The sharing mode of the virtual disk
+
+        """
+        sharing = disk.get('sharing')
```
`disk` is undefined.

cspeidel replied:
Hi @goneri. I noticed this last night when running the tests. I have since fixed, but not pushed it. I have this pull request marked as "draft", I will mark it ready for review and out of draft status when I'm done with the integration tests. Thanks for your initial review!

cspeidel replied:
Added the disk parameter and pushed in f3214482d6a202c74491de2286fb1185b1d3849c. Thanks!

cspeidel commented:
Hi @goneri, I've changed the PR to 'ready for review'. In addition to adding the integration tests, I've tested against an actual vSphere client 7.0.0.10000. Setting "sharing: True" results in multi-writer being set on the disk:
<img width="361" alt="Screen Shot 2020-06-02 at 5 26 23 PM" src="https://user-images.githubusercontent.com/19958998/83579646-125a3300-a4f7-11ea-8caa-ee4cae6d5bfc.png">

goneri commented:
@cspeidel looks good to me. Can you please add a changelog fragment? https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment

cspeidel commented:
Thank you @goneri, I've added the changelog fragment. I noticed some variations in the naming of the existing changelog files, but the documentation seemed to indicate I should use the issue number, followed by dashes in the description of the change, and ending in `.yaml` so that's what I went with. Do you normally squash the commits during the merge, or is that something I should do?

goneri commented:
Thanks for that. It looks good to me.

Yes, Zuul will actually Squash the commits and use the message of the first commit. So, if you want to keep a nice/clean commit message, I suggest you to rebase/squash your branch yourself.

cspeidel commented:
> Thanks for that. It looks good to me.
> 
> Yes, Zuul will actually Squash the commits and use the message of the first commit. So, if you want to keep a nice/clean commit message, I suggest you to rebase/squash your branch yourself.

Aha, ok I rebased so I'm up to date and I also squashed to edit my commit message to be similar to the format of past commit messages.

goneri commented:
@Akasurde can please you take a look at this PR? It looks ready to me.

goneri commented:
Thank you @cspeidel for your contribution! :-)
